### PR TITLE
Add DataFile and PCBObject

### PR DESCRIPTION
**Pull request by PiersonBro**

### CODE REVIEW

File: headers/DataFile.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+////////////////////////////////////////////////
```
Comment (tgamlem) [inline headers/DataFile.h]: We probably don't want this old file header in here

File: headers/PCBObject.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
+////////////////////////////////////////////////
+// Narrative.h
```
Comment (tgamlem) [inline headers/PCBObject.h]: Same thing here

File: .vscode/c_cpp_properties.json [hunk 3]
```diff
@@ -3,13 +3,14 @@
...
-                "\"C:/googletest/include\""
+                "${workspaceFolder}/**",
+                "/usr/local/Cellar/gcc/11.2.0_3/include/c++/11/tr1",
+                "/usr/local/Cellar/gcc/11.2.0_3/include/c++/11/x86_64-apple-darwin20"
             ],
             "defines": [
                 "_DEBUG"
             ],
-            "compilerPath": "\"C:/mingw-w64/mingw64/bin/g++.exe\"",
+            "compilerPath": "/usr/bin/clang++",
```
Comment (tgamlem) [inline .vscode/c_cpp_properties.json]: I think we might need to have this file outside of source control. I had to make some changes to get it running on windows

File: source/DataFile.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#include "DataFile.h"
+#include <cstdlib>
+#include <chrono>
+#include <sstream>
+
+DataFile::DataFile() {
+    // srandom();
+    int pid = random() % 30000 + 300;
```
Comment (tgamlem) [inline source/DataFile.cpp]: I had to change `random()` to `rand()` to get it compiling on windows. Does `rand()` work on Mac?

File: .vscode/c_cpp_properties.json (hunk 3 above)
Comment (PiersonBro) [reply .vscode/c_cpp_properties.json]: This file shouldn't affect how cmake works? Did you use vscode to compile. Regardless, it shouldn't be tracked.

**Comment by PiersonBro** [comment]
👀

File: .vscode/c_cpp_properties.json (hunk 3 above)
Comment (tgamlem) [reply .vscode/c_cpp_properties.json]: Nope, I was using cmake. Probably some weird disconnect between windows and mac

**Review by tgamlem — APPROVED** [review]

**Pull request merged**